main.go

```diff
@@ -9,19 +12,30 @@ import (
 	"main/repo"
 	"main/service"
 	"net/http"
-
-	"github.com/gorilla/mux"
-	"github.com/joho/godotenv"
 )
 
 func startServer(userHandler *handler.UserHandler, authHandler *authHandler.AuthHandler) {
 	router := mux.NewRouter().StrictSlash(true)
 
+	c := cors.New(cors.Options{
+		AllowedOrigins:   []string{"http://localhost:4200"},
+		AllowCredentials: true,
+		AllowedMethods: []string{
+			http.MethodGet,
+			http.MethodPost,
+			http.MethodPut,
+			http.MethodDelete,
+		},
+		AllowedHeaders: []string{
+			"*",
+		},
+	})
 	router.HandleFunc("/users", userHandler.CreateUser).Methods("POST")
 	router.HandleFunc("/users/{username}", authHandler.Authorize(userHandler.GetOne, "Admin")).Methods("GET")
 	router.HandleFunc("/login", authHandler.Login).Methods("POST")
+	r := c.Handler(router)
```

Comment:
Does this need to be done after defining handler functions. Will this put cors on all handlers before this line? Also can we do `r := c.handler(mux.NewRouter().StrictSlash(true))` so that we don't have to define another variable?

Reply:
We can't do that but we can just place `c.Handler(router)` in `log.Fatal(http.ListenAndServe(":3000", r))` like this `log.Fatal(http.ListenAndServe(":3000",c.Handler(router)))` so we don't have to define another variable.